### ISSUE TITLE
Support for `IN` sub-query expressions in `SQLContext`

### ISSUE THREAD

**Issue by csubhodeep** [pola-rs/polars#8918]
### Problem description

I wish Polars SQL supported `IN` and `NOT IN` sub-query expressions. For example, for the below snippet

```python
import polars as pl

test1 = pl.DataFrame({"a": [1,2,3], "b": ['x', 'y', 'z']})
test2 = pl.DataFrame({"c": [11,22,33], "d": ['x', 'w', 'z']})
sql_ctx = pl.SQLContext(register_globals=True)
res = sql_ctx.execute("""
SELECT * 
FROM test1
WHERE b IN (SELECT d FROM test2)
""")
print(res.collect())
```

I should get
```shell
shape: (2, 2)
┌─────┬─────┐
│ a   ┆ b   │
│ --- ┆ --- │
│ i64 ┆ str │
╞═════╪═════╡
│ 1   ┆ x   │
│ 3   ┆ z   │
└─────┴─────┘
```

References:
1. [Subquery Expressions](https://www.postgresql.org/docs/15/functions-subquery.html#FUNCTIONS-SUBQUERY-IN)

**Comment by holicc** [comment]
I'd like to implement this feature. Do you have any suggestions on how to do this?

**Comment by csubhodeep** [comment]
Unfortunately, I am not that fluent with Rust yet to be able to give you any viable suggestions. Maybe @alexander-beedie or @universalmind303 would be able to help you.

**Comment by alexander-beedie** [comment]
Closed by #11218 ✌️

**Comment by alexander-beedie** [comment]
So, subquery support now exists in general (hence closing the issue) but they're _are_ some minor gotchas that we can look at improving:

For example you currently need to qualify your `SELECT *` above, like so...
```
sql_ctx.execute("""
  SELECT test1.* 
  FROM test1
  WHERE b IN (SELECT d FROM test2)
""").collect()
```
...but we should really handle this better so it is unnecessary.
